owlkettle/guidsl.nim

```diff
@@ -45,6 +45,7 @@ type
       of NodeWidget:
         widget: seq[string]
         adder: Adder
+        widgetRefVar: Option[NimNode] # Contains a variable of type StateRef that may receive a reference to the constructed widget.
```

Comment:
```suggestion
        stateRef: NimNode
```

And all changes needed to make this compile. `stateRef.isNil` should be used instead of optionals.